### ISSUE TITLE
Worker always dies after change to catch CLD signal

### ISSUE THREAD

**Issue by AesSedai** [briandunn/flatware#84]
Hello, I've noticed that my test suite passes without error using flatware / flatware-rspec 2.2.1. When I upgraded to 2.3.1, my test suite now fails every run with a message that a worker has died, despite all the tests passing.

I can't share a full reproduction because this is a private codebase, but the gist is:
- ruby 2.6.10
- rspec 3.12.0
- bundler 2.4.16

This test suite doesn't utilize rails, so it only loads a `spec_helper.rb` and the `flatware_helper.rb`.

The output looks roughly as follows:
```
$ bundle exec rake test:suites:xyz:abc

Running specs for: xyz/abc
LD_LIBRARY_PATH=/usr/lib64 bundle exec flatware rspec -w 4
.......................................................
Finished in 0.05395 seconds (files took 1.41 seconds to load)
55 examples, 0 failures

A worker died!

Cleaning up. Please wait...

Finished in 0.05395 seconds (files took 1.41 seconds to load)
55 examples, 0 failures
done.
rake aborted!
```

I've tested removing the
```
::Signal.trap('CLD') { signals << :cld }
```
and the suite passes again. There isn't really any other info to go by, so I'm having trouble narrowing down what exactly is causing the `CLD` signal to fire. Perhaps you have some suggestions?

**Comment by mbell697** [comment]
We're seeing this as well. 

Doing a quick scan of the code changes it seems to me that the reason for `SIGCHILD` isn't being checked and `SIGCHILD` is sent for normal clean process exits as well. I suspect there may be a race between the signal and happy path process exiting code.

**Comment by jdelStrother** [comment]
Yeah, I see the same problem. CLD is received for each worker-exit (even successfully completed ones), so it doesn't seem like that should trigger the "A worker died" message.

**Comment by briandunn** [comment]
🤔 great point. thanks for the issue. I'll look into adding a check that the worker exited early, as apposed to naturally.

**Comment by briandunn** [comment]
@AesSedai I've tried to address this issue in #89. Could you test that branch and see if it's working for you?

**Comment by AesSedai** [comment]
Hi @briandunn, I'm out of pocket this week but can try testing it this coming Monday. Thanks.

**Comment by jdelStrother** [comment]
FWIW #89 works well for me

**Comment by briandunn** [comment]
@jdelStrother  thanks for trying it out! Merged #89 and released a patch. Optimistically closing.

**Comment by AesSedai** [comment]
Hi @briandunn, I was able to re-test this morning and it runs cleanly. Thank you!
